gcp/app-engine/main.py

```diff
@@ -7,15 +8,39 @@
 import sqlalchemy
 from google.cloud import secretmanager
 import pg8000
+import firebase_admin
+from firebase_admin import auth
 
 app = Flask(__name__)
 
+firebase_admin.initialize_app()
+
 DEFAULT_PROJECT_ID = 'stoked-depth-428423-j7'
 DEFAULT_VERSION_ID = 'latest'
 
 PUBLIC_API_PREFIX = '/api/v1'
 PRIVATE_API_PREFIX = '/protected/api/v1'
 
+# Decorator to parse auth token and extract user email
```

Comment:
decorator to (1) parse auth header, (2) throw 401 if missing or not valid, and (3) extract the user_email field and pass it into the flask callback